### PR TITLE
Daman banner update

### PR DESCRIPTION
**Pull request by damanpreet97**
Includes changes to banner component

### CODE REVIEW

File: learn/src/main/java/org/navgurukul/learn/courses/db/models/BaseCourseContent.kt [hunk 1]
```diff
@@ -171,4 +177,8 @@ enum class CodeType {
 
 enum class DecorationType {
     number, bullet
+}
+
+enum class MerakiButtonType {
```
Comment (bsroomy) [inline learn/src/main/java/org/navgurukul/learn/courses/db/models/BaseCourseContent.kt:182]: This should be moved to meraki button class

File: learn/src/main/java/org/navgurukul/learn/ui/common/MerakiButton.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import org.navgurukul.commonui.resources.ResourceResolver
+import org.navgurukul.learn.R
+import org.navgurukul.learn.courses.db.models.BannerAction
+import org.navgurukul.learn.courses.db.models.MerakiButtonType
+
+class MerakiButton @JvmOverloads constructor(
+    context: Context,
+    attributes: AttributeSet? = null,
+    defaultStyleAttr: Int = 0,
+) : FrameLayout(context, attributes, defaultStyleAttr) {
```
Comment (bsroomy) [inline learn/src/main/java/org/navgurukul/learn/ui/common/MerakiButton.kt:20]: The parent should be a button, framelayout is useless here

File: learn/src/main/java/org/navgurukul/learn/ui/common/MerakiButton.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    private var merakiButton: MaterialButton
+
+    init {
+        inflate(context, R.layout.layout_meraki_button, this)
+        merakiButton = findViewById(R.id.merakiButton)
+    }
+
+    fun setData(element:BannerAction, action: (BannerAction) -> Unit){
+        merakiButton.text = element.label?:"Default"
```
Comment (bsroomy) [inline learn/src/main/java/org/navgurukul/learn/ui/common/MerakiButton.kt:30]: This method can be added as extension outside this class or simply a method in the class where this method is used

File: learn/src/main/java/org/navgurukul/learn/ui/common/MerakiButton.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            merakiButton.iconGravity = ICON_GRAVITY_END
+
+        }?:run { merakiButton.icon = null }
+
+        element.variant?.let {
+            setVariant(it)
+        }
+    }
+
+    fun setVariant(variant: MerakiButtonType){
```
Comment (bsroomy) [inline learn/src/main/java/org/navgurukul/learn/ui/common/MerakiButton.kt:51]: variant should be a property with a getter and a setter (could be kotlin getter setters)

File: learn/src/main/java/org/navgurukul/learn/courses/db/models/BaseCourseContent.kt [hunk 5]
```diff
@@ -154,7 +156,11 @@ data class BannerAction(
         @Json(name = "label")
         var label: String?,
         @Json(name = "data")
-        var data: String?
+        var data: String?,
+        @Json(name = "icon")
+        var icon: String? = "ic_arrow_right",
```
Comment (bsroomy) [inline learn/src/main/java/org/navgurukul/learn/courses/db/models/BaseCourseContent.kt:161]: Default value should be removed

File: learn/src/main/res/layout/layout_meraki_button.xml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (bsroomy) [inline learn/src/main/res/layout/layout_meraki_button.xml:2]: This whole file can be removed by making material button as parent of meraki button class

**Review by bsroomy — CHANGES_REQUESTED** [review]

**Review by abhishekgupta92 — APPROVED** [review]

**Pull request merged**